package/MDAnalysis/coordinates/MMTF.py

```diff
@@ -0,0 +1,62 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4 
+#
+# MDAnalysis --- http://www.MDAnalysis.org
+# Copyright (c) 2006-2015 Naveen Michaud-Agrawal, Elizabeth J. Denning, Oliver
+# Beckstein and contributors (see AUTHORS for the full list)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+
+"""MMTF trajectory reader --- :mod:`MDAnalysis.coordinates.MMTF`
+================================================================
+
+Reads coordinates data from the `Macromolecular Transmission Format
+(MMTF) format`_.  This should generally be a quicker alternative to PDB.
+
+.. versionadded:: 0.16.0
+
+Classes
+-------
+
+.. autoclass:: MMTFReader
+   :members:
+
+
+.. _Macromolecular Transmission Format (MMTF) format: https://mmtf.rcsb.org/
+"""
+
+import mmtf
+
+from . import base
+
+
+def _parse_mmtf(fn):
+    if fn.endswith('gz'):
+        return mmtf.parse_gzip(fn)
+    else:
+        return mmtf.parse(fn)
+
+
+class MMTFReader(base.SingleFrameReader):
+    format = 'MMTF'
+
+    def _read_first_frame(self):
+        # TOOD: Check units?
+        top = _parse_mmtf(self.filename)
+        self.n_atoms = top.num_atoms
+
+        self.ts = ts = self._Timestep(self.n_atoms,
+                                      **self._ts_kwargs)
+        ts._pos[:, 0] = top.x_coord_list.copy()
```

Comment:
Really? When I do the following, the entries in the slice are copied:
```python
import numpy as np
coord = np.zeros((20, 3), dtype=np.float32)
x = np.arange(len(coord), dtype=np.float32)
x0 = x.copy()  # for comparison later

# assign slice
coord[:, 0] = x
assert np.all(coord[:, 0] == x)

# change x in place
x *= -1
# if you are correct then the next assert fails
assert np.all(coord[:, 0] == x0)

# you say that, the slice became -x0 ... it does not for me
assert np.all(x == -x0)
assert np.all(coord[:, 0] == -x0)
```

(I do remember we had some discussions about copy in Timestep a while ago and I cannot remember the context. But for this simple example, slice assignment seems to do the copy.)

The copy does have a measurable impact (for 1M coordinates):
```python
In [22]: coord = np.zeros((int(1e6), 3), dtype=np.float32)
In [23]: x = np.arange(len(coord), dtype=np.float32)
In [24]: %timeit coord[:, 0] = x
1000 loops, best of 3: 1.62 ms per loop
In [25]: %timeit coord[:, 0] = x.copy()
100 loops, best of 3: 3.14 ms per loop
```